ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -30,5 +30,266 @@
 #include <catch2/catch.hpp>
 
 #include "gcp_connection.h"
+#include <typeinfo>
 
-TEST_CASE("Testing class gcpConnection", "gcp-connection") {}
+namespace gcs = google::cloud::storage;
+using namespace gcs;
+
+static std::string
+create_file(const std::string file_name, const std::string payload)
+{
+    std::ofstream file(file_name);
+    file << payload;
+    file.close();
+    return file_name;
+}
+
+static auto
+upload_file(gcs::Client client, std::string bucket_name, const std::string bucket_prefix,
+  const std::string file_name, const std::string object_name)
+{
+    auto metadata = client.UploadFile("./" + file_name, bucket_name, bucket_prefix + object_name);
+
+    return metadata;
+}
+
+static bool
+check_exist(gcs::Client client, std::string bucket_name, const std::string bucket_prefix,
+  const std::string object_name)
+{
+    auto metadata = client.GetObjectMetadata(bucket_name, bucket_prefix + object_name);
+
+    // Metadata ok implies that the file is present.
+    if (metadata.ok()) {
+        return true;
+    }
+
+    return false;
+}
+
+static bool
+check_object_num(
+  gcs::Client client, std::string bucket_name, const std::string bucket_prefix, const int num)
+{
+    auto objects_iterator = client.ListObjects(bucket_name, gcs::Prefix(bucket_prefix));
+    return std::distance(objects_iterator.begin(), objects_iterator.end()) == num;
+}
+
+// Concatenates a random suffix to the prefix being used for the test object keys. Example of
+// generated test prefix: "gcptest/unit/2022-31-01-16-34-10/623843294--".
+static std::string
+randomize_test_prefix(std::string prefix)
+{
+    char time_str[100];
+    std::time_t t = std::time(nullptr);
+
+    REQUIRE(std::strftime(time_str, sizeof(time_str), "%F-%H-%M-%S", std::localtime(&t)) != 0);
+
+    prefix += time_str;
+
+    // Create a random device and use it to generate a random seed to initialize the generator.
+    std::random_device my_random_device;
+    unsigned seed = my_random_device();
+    std::default_random_engine my_random_engine(seed);
+
+    prefix += '/' + std::to_string(my_random_engine());
+    prefix += "--";
+
+    return prefix;
+}
+
+TEST_CASE("Testing class gcpConnection", "gcp-connection")
+{
+    std::string bucket_name = "unit_testing_gcp";
+    std::string prefix = "gcptest-unit-";
+
+    // Setup the test environment.
+    std::string bucket_prefix = randomize_test_prefix(prefix);
+    gcp_connection conn(bucket_name, bucket_prefix);
+
+    const std::string object_name = "test_object";
+    const std::string file_name = object_name + ".txt";
+    const std::string non_existant_object_name = "test_non_exist";
+    const std::string non_existant_file_name = non_existant_object_name + ".txt";
+    const bool list_single = true;
+    std::vector<std::string> objects;
+
+    auto client = gcs::Client();
```

Comment:
np: fix this to not use auto please

Reply:
addressed in [2ace538](https://github.com/wiredtiger/wiredtiger/pull/8749/commits/2ace538bb48cef22cdd59a28392f5f0ea56b8396)